File: sdk/src/test/java/com/cxense/cxensesdk/model/ImpressionTest.java
package com.cxense.cxensesdk.model;

import com.cxense.cxensesdk.BaseTest;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class ImpressionTest extends BaseTest {
    private Impression impression;

    @Before
    public void setUp() throws Exception {
        super.setUp();
        impression = new Impression("url", 1);
    }

    @Test
    public void getClickUrl() {
        String url = "clickUrl";
        impression.clickUrl = url;
        assertEquals(url, impression.getClickUrl());
    }

    @Test
    public void getSeconds() {
        int seconds = 12345;
        impression.seconds = seconds;
        assertEquals(seconds, impression.getSeconds());
    }
}

File: sdk/src/test/java/io/piano/android/cxense/model/ImpressionTest.kt
package io.piano.android.cxense.model

import io.piano.android.cxense.assertFailsWithMessage
import kotlin.test.Test

class ImpressionTest {
    @Test
    fun createValid() {
        Impression("url", 5)
    }

    @Test
    fun createWithEmptyUrl() {
        assertFailsWithMessage<IllegalArgumentException>("clickUrl", "Expected fail for clickUrl") {
            Impression("", 5)
        }
    }

    @Test
    fun createWithNegativeSeconds() {
        assertFailsWithMessage<IllegalArgumentException>("Seconds", "Expected fail for seconds") {
            Impression("url", -5)
        }
    }
}
